manifests/init.pp

```diff
@@ -30,66 +30,183 @@
-  windowsfeature { 'Web-WebServer':
-  installmanagementtools => true,
-  } ->
-  windowsfeature { 'Web-Asp-Net45':
-  } ->
+  # added compatibility mode so module doesn't break for previous users
+  if $compatibility_mode {
+    # add windows features
+    windowsfeature { 'Web-WebServer':
+      installmanagementtools => true,
+    } ->
+    windowsfeature { 'Web-Asp-Net45':
+    } ->
 
-  # remove default web site
-  iis::manage_site {'Default Web Site':
-    ensure        => absent,
-    site_path     => 'any',
-    app_pool      => 'DefaultAppPool'
-  } ->
+    # remove default web site
+    if $disable_default_website {
+      iis::manage_site {'Default Web Site':
+        ensure    => stopped,
+        app_pool  => 'DefaultAppPool',
+      }
+    }
 
-  # application in iis
-  iis::manage_app_pool { "${_chocolatey_server_app_pool_name}":
-    enable_32_bit           => true,
-    managed_runtime_version => 'v4.0',
-  } ->
-  iis::manage_site {'chocolatey.server':
-    site_path     => $_chocolatey_server_location,
-    port          => "${_chocolatey_server_app_port}",
-    ip_address    => '*',
-    app_pool      => "${_chocolatey_server_app_pool_name}",
-  } ->
+    # application in iis
+    iis::manage_app_pool { "${_chocolatey_server_app_pool_name}":
+      enable_32_bit           => true,
+      managed_runtime_version => 'v4.0',
+    } ->
+
+    iis::manage_site {'chocolatey.server':
+      site_path  => $_chocolatey_server_location,
+      port       => "${_chocolatey_server_app_port}",
+      ip_address => '*',
+      app_pool   => "${_chocolatey_server_app_pool_name}",
+    }
+
+  } else {
+    # if not using compatibility mode, then use newer methods to setup windows
+    # features and iis
+    # this removes dependencies from the windowsfeature and iis module but adds
+    # a dependency for:
+    #   puppetlabs-dsc
+    #   puppetlabs-powershell
+    #   puppetlabs-reboot
+    #   puppetlabs-stdlib
+
+    # install required windows features
+    dsc_windowsfeature{ 'Install-Web-WebServer':
+      dsc_ensure => present,
+      dsc_name   => 'Web-WebServer',
+    }
+
+    dsc_windowsfeature{ 'Install-Web-Asp-Net45':
+      dsc_ensure => present,
+      dsc_name   => 'Web-Asp-Net45',
+    }
+
+    dsc_windowsfeature{ 'Install-Web-Mgmt-Tools':
+      dsc_ensure => present,
+      dsc_name   => 'Web-Mgmt-Tools',
+    }
+
+    if $disable_default_website {
+      dsc_xwebsite{ 'Default_Web_Site':
+        dsc_ensure       => 'present',
+        dsc_name         => 'Default Web Site',
+        dsc_physicalpath => $default_website_path,
+        dsc_state        => 'stopped',
+        require          => Dsc_windowsfeature['Web-WebServer'],
+      }
+    }
+
+    # setup app pool for website
+    dsc_xwebapppool { "${_chocolatey_server_app_pool_name}":
+      dsc_ensure                => present,
+      dsc_name                  => "${_chocolatey_server_app_pool_name}",
+      dsc_enable32bitapponwin64 => false,
```

Comment:
I tested this again, and it looks like it does work as-is, so I removed the change for disabling the 32-bit app_pool mode.